libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -421,15 +465,57 @@ bool AP_Arming::battery_checks(bool report)
         }
 
         for (uint8_t i = 0; i < _battery.num_instances(); i++) {
+            // voltage check
             if ((_min_voltage[i] > 0.0f) && (_battery.voltage(i) < _min_voltage[i])) {
                 if (report) {
-                    gcs().send_text(MAV_SEVERITY_CRITICAL, "PreArm: Battery %d voltage %.1f below minimum %.1f",
+                    gcs().send_text(MAV_SEVERITY_CRITICAL, "PreArm: Battery %u voltage %.1f below min %.1f",
                             i+1,
                             (double)_battery.voltage(i),
                             (double)_min_voltage[i]);
                 }
                 return false;
             }
+
+            float temperature = AP_ARMING_BATTERY_DEFAULT_CELSIUS_TEMPERATURE;
+            // temperature minimum check
+            if (static_cast<int16_t>(_min_temperature[i] * 10) > static_cast<int16_t>(AP_ARMING_BATTERY_DEFAULT_CELSIUS_TEMPERATURE * 10)) {
```

Comment:
Why is everything being scaled by 10 here? (and on all the other comparison lines below). All the casting is unneeded as well as the AP_Param's all handle the comparisons to their base types correctly.

Reply:
I saw the values below the decimal point are irrational numbers. I judged with an integer value by 10 times. I will log.